pandas/core/groupby.py

```diff
@@ -642,9 +642,20 @@ def apply(self, func, *args, **kwargs):
 
         func = self._is_builtin_func(func)
 
-        @wraps(func)
-        def f(g):
-            return func(g, *args, **kwargs)
+        # this is needed so we don't try and wrap strings. If we could
+        # resolve functions to their callable functions prior, this
+        # wouldn't be needed
+        if args or kwargs:
+            if is_callable(func):
+
+                @wraps(func)
+                def f(g):
+                    return func(g, *args, **kwargs)
+            else:
+                raise ValueError('func must be a callable if args or '
+                                 'kwargs are supplied')
+        else:
+            f = func
```

Comment:
I am all consolidating things a bit, sure ``_is_built_in_func`` could just return wrappers; have to be actual functions though (e.g. only return a wrapper if its a function that is defined and NOT a callable already).

Its done but might be done in several places.

You might want to do that in another PR. (as these types of conversions happen in groupby,resample,window functions); though did try to make it all in one place.